src/screens/WebsocketResearchScreen/SampleChatScreen.tsx

```diff
@@ -68,34 +82,47 @@ const SampleChatScreen = () => {
     flatlistRef.current?.scrollToEnd();
   };
 
+  React.useEffect(() => {
+    return () => {
+      clearReplyPreview();
+    };
+  }, []);
+
   return (
-    <View style={styles.keyboardAvoidingView}>
-      {selectedChannel ? (
-        <ChatDetailHeader
-          onAvatarPress={goToChatInfoScreen}
-          onBackPress={goBackFromChatScreen}
-          onThreeDotPress={goToChatInfoScreen}
-          avatar={selectedChannel?.channelPicture}
-          user={selectedChannel?.name}
-          type={ANONYMOUS}
+    <>
+      <StatusBar barStyle="light-content" backgroundColor={colors.anon_primary} />
+
+      <View style={styles.keyboardAvoidingView}>
+        {selectedChannel ? (
+          <ChatDetailHeader
+            onAvatarPress={goToChatInfoScreen}
+            onBackPress={goBackFromChatScreen}
+            onThreeDotPress={goToChatInfoScreen}
+            avatar={selectedChannel?.channelPicture}
+            user={selectedChannel?.name}
+            type={ANONYMOUS}
+          />
+        ) : null}
+        <FlatList
+          contentContainerStyle={styles.flatlistContainer}
+          style={styles.chatContainer}
+          data={updatedChats}
+          inverted={true}
+          initialNumToRender={20}
+          alwaysBounceVertical={false}
+          bounces={false}
+          onLayout={scrollToEnd}
+          keyExtractor={(item, index) => item?.id || index.toString()}
+          renderItem={renderChatItem}
         />
-      ) : null}
-      <FlatList
-        contentContainerStyle={styles.contentContainerStyle}
-        style={styles.chatContainer}
-        data={updatedChats}
-        inverted={true}
-        initialNumToRender={20}
-        alwaysBounceVertical={false}
-        bounces={false}
-        onLayout={scrollToEnd}
-        keyExtractor={(item, index) => item?.id || index.toString()}
-        renderItem={renderChatItem}
-      />
-      <View style={styles.inputContainer}>
-        <AnonymousInputMessage onSendButtonClicked={sendChat} type={ANONYMOUS} />
+
+        {replyPreview && <ChatReplyPreview type={ANONYMOUS} />}
+        <View style={styles.inputContainer}>
+          <AnonymousInputMessage onSendButtonClicked={sendChat} type={ANONYMOUS} />
+        </View>
+        <Loading visible={loading} />
       </View>
-    </View>
+    </>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this section, you've added a conditional rendering for `ChatReplyPreview` based on `replyPreview`. This is a good practice for handling optional components. However, ensure that the `ChatReplyPreview` component properly handles all possible states of its props to avoid potential crashes or bugs. 

Also, you've introduced a loading state with `setLoading` and `loading`. Make sure to set and reset this state at appropriate times during your chat operations to accurately reflect the loading status.


<!-- This is an auto-generated comment by OSS CodeRabbit -->